src/sentry/tasks/integrations/github/open_pr_comment.py

```diff
@@ -158,34 +159,44 @@ def safe_for_comment(
                 tags={"type": GithubAPIErrorType.UNKNOWN.value, "code": e.code},
             )
             logger.exception("github.open_pr_comment.unknown_api_error", extra={"error": str(e)})
-        return False
+        return False, []
 
     safe_to_comment = True
-    if pullrequest_resp["state"] != "open":
-        metrics.incr(
-            OPEN_PR_METRICS_BASE.format(key="rejected_comment"), tags={"reason": "incorrect_state"}
-        )
-        safe_to_comment = False
-    if pullrequest_resp["changed_files"] > OPEN_PR_MAX_FILES_CHANGED:
-        metrics.incr(
-            OPEN_PR_METRICS_BASE.format(key="rejected_comment"), tags={"reason": "too_many_files"}
-        )
-        safe_to_comment = False
-    if pullrequest_resp["additions"] + pullrequest_resp["deletions"] > OPEN_PR_MAX_LINES_CHANGED:
-        metrics.incr(
-            OPEN_PR_METRICS_BASE.format(key="rejected_comment"), tags={"reason": "too_many_lines"}
-        )
-        safe_to_comment = False
-    return safe_to_comment
 
+    changed_file_count = 0
+    changed_lines_count = 0
+    filtered_pr_files = []
 
-def get_pr_filenames(
-    gh_client: GitHubAppsClient, repository: Repository, pull_request: PullRequest
-) -> List[str]:
-    pr_files = gh_client.get_pullrequest_files(repo=repository.name, pull_number=pull_request.key)
+    for file in pr_files:
+        filename = file["filename"]
+        # don't count the file if it was added or is not a Python file
+        if file["status"] == "added" or not filename.endswith(".py"):
+            continue
+
+        changed_file_count += 1
+        changed_lines_count += file["changes"]
+        filtered_pr_files.append(file)
+
+        if changed_file_count > OPEN_PR_MAX_FILES_CHANGED:
+            metrics.incr(
+                OPEN_PR_METRICS_BASE.format(key="rejected_comment"),
+                tags={"reason": "too_many_files"},
+            )
+            return False, []
```

Comment:
rather than returning false/empty json, any reason why we don't just allow comments up to the limit?

Reply:
the response returns up to 30 files. we can early exit out of the loop if there are too many files or too many lines in the PR before we iterate through all 30

Reply:
actually... i'm not exactly sure what you mean by "up to the limit"